warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java

```diff
@@ -394,7 +394,7 @@ else if (ExceededValueThresholdMarkerJexlNode.instanceOf(node) || ExceededOrThre
         else if (ASTDelayedPredicate.instanceOf(node) || ASTEvaluationOnly.instanceOf(node)) {
             if (isNoFieldOnly(node)) {
                 state = STATE.IGNORABLE;
-            } else if (isNonEvent(node)) {
+            } else if (ASTEvaluationOnly.instanceOf(node) && isNonEvent(node)) {
```

Comment:
Should we add a check for delayed here as well?

Reply:
a nonEvent delayed is exactly what we can now process